src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -787,11 +803,41 @@ public void setHostKeyVerificationStrategy(HostKeyVerificationStrategyEnum hostK
     public HostKeyVerificationStrategyEnum getHostKeyVerificationStrategy() {
         return hostKeyVerificationStrategy != null ? hostKeyVerificationStrategy : HostKeyVerificationStrategyEnum.CHECK_NEW_SOFT;
     }
+
+    @CheckForNull
+    public String getAmiOwners() {
+        return amiOwners;
+    }
+
+    @DataBoundSetter
+    public void setAmiOwners(String amiOwners) {
+        this.amiOwners = amiOwners;
+    }
+
+    @CheckForNull
+    public String getAmiUsers() {
+        return amiUsers;
+    }
+
+    @DataBoundSetter
+    public void setAmiUsers(String amiUsers) {
+        this.amiUsers = amiUsers;
+    }
+
+    @CheckForNull
+    public List<EC2Filter> getAmiFilters() {
+        return amiFilters;
+    }
+
+    @DataBoundSetter
+    public void setAmiFilters(List<EC2Filter> amiFilters) {
+        this.amiFilters = amiFilters;
+    }
     
     @Override
     public String toString() {
         return "SlaveTemplate{" +
-                "ami='" + ami + '\'' +
+                "description='" + description + '\'' +
```

Comment:
why this change? I don't see the relationship with the desired fix and I'd appreciate if you could elaborate. Thanks!

Reply:
Because the image ID can be dynamic are this PR. In order to list the image ID, you'd need to do a round trip to AWS every time this is called. Which is actually often because `logProvisionInfo` uses it.

Reply:
To expand on that a bit more, you could just leave it as is but in the case where you were using filters it would just say `ami=""`, which wouldn't be helpful if you were looking in the logs. I actually think this is an improvement generally since you could have multiple templates sharing the same image ID. This way you can differentiate them if you use a unique description for each template.